src/Common/MoistThermodynamics/states.jl

```diff
@@ -201,3 +201,35 @@ function LiquidIcePotTempSHumNonEquil_given_pressure(θ_liq_ice::FT, p::FT, q_pt
     e_int = internal_energy(T, q_pt)
     return PhaseNonEquil(e_int, ρ, q_pt)
 end
+
+"""
+    guaranteed_convergence_range(FT)
+
+A range of input arguments to thermodynamic state constructors
+ - `e_int`
+ - `ρ`
+ - `q_tot`
+ - `q_pt`
+ - `T`
+that are guaranteed converge in saturation adjustment.
+
+Note that the output vectors are of size ``n^3``.
+"""
+function guaranteed_convergence_range(FT, n)
+  ρ_range  = range(FT(1e-3),  stop=FT(1.6), length=n)
+  RH_range = range(FT(0),     stop=FT(1  ), length=n)
```

Comment:
```suggestion
  RH_range = range(FT(0),     stop=FT(1.1  ), length=n)
```
You'd want some RH > 1 here (otherwise you do not need saturation adjustment because air is unsaturated).